peps/pep-0757.rst

```diff
@@ -49,11 +49,15 @@ Specification
 Layout API
 ----------
 
-Data needed by `GMP <https://gmplib.org/>`_-like import-export functions.
+Data needed by `GMP <https://gmplib.org/>`_-like `import
+<https://gmplib.org/manual/Integer-Import-and-Export#index-mpz_005fimport>`_-`export
+<https://gmplib.org/manual/Integer-Import-and-Export#index-mpz_005fexport>`_
+functions.
 
 .. c:struct:: PyLongLayout
 
-   Layout of an array of digits, used by Python :class:`int` object.
+   Layout of an array of digits, used by Python :class:`int` object to
+   represent "big enough" integers.
```

Comment:
"big enough"? Is there something more specific that we could use?

Reply:
Perhaps, "that not fits in some machine-sized integer type".  But maybe even that is a some constraint for future changes.

I would prefer here some vague wording, that tell us - it's all about asymptotic representation of int's, i.e. for some implementation-dependent value it will switch to represent integers as "array of digits" in base-``PyLong_BASE`` positional system.

Reply:
I would recommend either removing the new wording that includes "big enough" or 

```suggestion
   represent an integer in an asymptotically fast format.
```

Reply:
No, it's not about being fast or slow.

Just at some value (currently, just starting from ``0``), the internal representation for int's switched to the array of unsigned "digits" (base-n positional system, where ``n`` is ``PyLong_Base = 1<<sys.int_info.bits_per_digit``), holding the absolute value (sign of the integer - stored separately).

This is somewhat a common denominator for all arbitrary precision libraries.  But parameters for such representation might be different.  PyLong_GetNativeLayout() returns one, valid for the given Python implementation.

Reply:
@skirpichev Thanks for somewhat clarifying. I think that I may not be clearly communicating my objection to using "big enough" integers. It's unclear what "big enough" means. 

@vstinner @encukou @pitrou, I would appreciate if you can take a look at this. I'm uncomfortable with "big enough" being documentation for this struct. As a reader, I'm trying to get a sense of when this is used and when it is not. I'm -1 on leaving "big enough" as it's not clear to me as a PEP Editor what this means, and I would like more explicit prose.